hrjobcontract/js/src/job-contract/controllers/modal/modal-contract.controller.js

```diff
@@ -765,6 +796,35 @@ define([
           $rootScope.options.pay[optionName] = _.mapValues(_.keyBy(data, 'value'), 'label');
         });
     }
+
+    /**
+     * Loads updated record of providers option group
+     *
+     * @param {Array} providers
+     * @returns {Promise}
+     */
+    function loadProviderOptions (providers) {
```

Comment:
you have all duplicated logic here, isn't there a way to avoid it

Reply:
Will it be advisable to make utilService return the required format (the only draw back is this will enforce other implementation using the getOptionValues to use the same format)? I will need your advice on how best to resolve this.